hutch_python/tests/test_namespace.py

```diff
@@ -66,9 +66,31 @@ def test_tree_namespace():
     assert xpp.sb2.obj5 == 5
 
 
+def test_same_name():
+    logger.debug('test_same_name')
+    # This should be ok, but make sure the warning is covered
+    objs = dict(hutch_stand=SimpleNamespace(dev=1),
+                hutch_stand_dev=2)
+    tree_namespace(objs=objs)
+
+
 def test_conflicting_name():
     logger.debug('test_conflicting_name')
-    # This should be ok, but make sure the warning is covered
-    scope = SimpleNamespace(hutch_stand=SimpleNamespace(dev=1),
-                            hutch_stand_dev=2)
-    tree_namespace(scope=scope)
+    # There was a bug where names could conflict with eachother
+    objs = dict(daq=Signal(name='daq'),
+                daq_repeat_echo='daq_repeat_echo',
+                daq_repeat=Signal(name='daq_repeat'),
+                daq_deep='daq_deep',
+                daq_deep_deep=Signal(name='daq_deep_deep'))
+    ns = tree_namespace(objs=objs)
+    assert ns.daq.name == 'daq'
+    assert ns.daq.repeat.echo == 'daq_repeat_echo'
+    assert ns.daq.repeat.name == 'daq_repeat'
```

Comment:
I rather like the tree concept in general, but this portion scares me a bit. Effectively, since `daq` exists in the top-level namespace, we are monkey-patching on additional attributes (`repeat` and `deep` here).

If I call `tree_namespace`, I wouldn't expect it to be modifying _any_ of the objects I passed, merely put them in a logical grouping. At the very least, that should be well-documented.

How common is this scenario where you have `obj` and many similarly named `obj_{...}` in a single environment?

Reply:
I hope this is rare. I also do not like the patching, but I don't *really* know what else to do about it. Ideal scenarios would be one of the following:
1. Names are very consistent (e.g. all_with_four_sections) and we error/warn instead of monkey-patching
2. There is some way to make a proxy object tree without doing any setattrs on the real objects

I'm at a bit of a loss of how to do this correctly